pallets/tickets/src/tests.rs

```diff
@@ -1 +1,600 @@
-// TODO
+use crate::mock;
+use crate as pallet_tickets;
+
+use crate::*;
+
+use crate::dummy_data::*;
+
+use crate::mock::*;
+
+use serde_json::json;
+
+use frame_system::offchain::{
+    SignedPayload, SigningTypes,
+};
+
+use frame_support::{
+    assert_noop, assert_ok,
+    traits::{
+        TypedGet,
+    }
+};
+
+use frame_support::dispatch::DispatchResult;
+
+use sp_core::{
+
+    H160,
+
+    H256,
+    
+    offchain::{
+        testing,
+    }
+};
+
+use sp_runtime::offchain::storage::StorageValueRef;
+
+use sp_runtime::SaturatedConversion;
+
+use codec::Encode;
+
+use ethabi::Token;
+
+pub use link_tests::link_ as link_;
+pub use internal_lock_update_tests::lock_ as lock_;
+
+mod link_tests {
+
+
+    use crate::FragUsage;
+
+    use super::*;
+
+    pub fn link_(signer: <Test as frame_system::Config>::AccountId, link_signature: &sp_core::ecdsa::Signature) -> DispatchResult {
+        TicketsPallet::link(
+            Origin::signed(signer),
+            link_signature.clone()
+        )
+    }
+
+    #[test]
+    fn link_should_work() {
+
+        new_test_ext().execute_with(|| {
+
+			let dd = DummyData::new();
+
+            let link = dd.link;
+
+
+            assert_ok!(
+                link_(link.clamor_account_id, &link.get_link_signature())
```

Comment:
Replace 
```rust
link_(...)
```
with
```rust
_ = link_(...)
```
to fix the annoying warning from compiler due to unhandled returned value.

Reply:
@aang114 generally speaking, if you use rust-analyzer in VSCode you will see immediately all the warnings. Very handy.

Reply:
Oh you were asking about rust-analyzer earlier because of this 😅 . 

Yeah I use it, but I never considered warnings important 😅. Will be mindful of them from now on

Reply:
@aang114 warnings are red flags, get rid of them ASAP.